crates/autopilot/src/solvable_orders.rs

```diff
@@ -170,29 +174,62 @@ impl SolvableOrdersCache {
     pub async fn update(&self, block: u64) -> Result<()> {
         let start = Instant::now();
         let min_valid_to = now_in_epoch_seconds() + self.min_order_validity_period.as_secs() as u32;
-        let db_solvable_orders = self.persistence.solvable_orders(min_valid_to).await?;
 
-        let mut counter = OrderFilterCounter::new(self.metrics, &db_solvable_orders.orders);
+        // A new auction should be created regardless of whether new solvable orders are
+        // found. The incremental solvable orders cache updater should only be
+        // enabled after the initial full SQL query
+        // (`persistence::solvable_orders`) returned some orders. Until then, `MIN_UTC`
+        // is used to indicate that no orders have been found yet by
+        // (`persistence::solvable_orders`). This prevents situations where
+        // starting the service with a large existing DB would cause
+        // the incremental query to load all unfiltered orders into memory, potentially
+        // leading to OOM issues.
+        let (db_solvable_orders, previous_creation_timestamp) = {
+            let lock = self.cache.lock().await;
+            match &*lock {
+                Some(cache) if cache.last_order_creation_timestamp > DateTime::<Utc>::MIN_UTC => (
+                    self.persistence
+                        .solvable_order_after(
```

Comment:
Here we are locking the cache for the duration of the function `solvable_order_after`